.infrastructure/helm-chart/todoapp/templates/deployment.yml

```diff
@@ -0,0 +1,86 @@
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: {{.Chart.Name}}
+  namespace: {{.Values.todoapp.namespace}}
+spec:
+  strategy:
+    type: RollingUpdate
+    rollingUpdate:
+      maxSurge: {{.Values.todoapp.rollingUpdate.maxSurge}}
+      maxUnavailable: {{.Values.todoapp.rollingUpdate.maxUnavailable}}
```

Comment:
Ensure that `{{.Values.todoapp.rollingUpdate.maxSurge}}` and `{{.Values.todoapp.rollingUpdate.maxUnavailable}}` are defined in your `values.yaml` file. If they're not defined, it could lead to deployment strategy issues.